mount/mount_unix_test.go

```diff
@@ -230,3 +233,47 @@ func TestRecursiveUnmountTooGreedy(t *testing.T) {
 		t.Fatal("expected dir-other to be mounted, but it's not")
 	}
 }
+
+func TestRecursiveUnmount_SubMountFailsToUnmount(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skip("root required")
+	}
+
+	tmp := t.TempDir()
+
+	dir := "sub1/sub2/sub3"
+	err := os.MkdirAll(path.Join(tmp, dir), 0o700)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	grandchild := path.Join(tmp, dir)
+	child := filepath.Dir(grandchild)
+	parent := filepath.Dir(child)
```

Comment:
This seems making things more complicated than needed to find paths we already know what they'll be (relative to `tmp` that is). Given that this only runs on unix/linux, we don't strictly need `filepath` (path separator will always be `/`) and because these paths are "fixed", we don't need to care about path-escaping etc. (also `grandchild` should probably be `grandChild` to be properly camelCase); 

perhaps something like:

```go
var (
	tmp        = t.TempDir()
	parent     = tmp + "/sub1"
	child      = tmp + "/sub1/sub2"
	grandChild = tmp + "/sub1/sub2/sub3"
)

err := os.MkdirAll(grandChild, 0o700)
if err != nil {
	t.Fatal(err)
}
```

That way, it's obvious at a glance what paths we're dealing with (the `var(    )` block makes them align nicely, so it's easier to read for this).